src/hooks/useRowSelection.ts

```diff
@@ -0,0 +1,31 @@
+import { createContext, useContext } from 'react';
+import type { SelectRowEvent } from '../types';
+
+const RowSelectionContext = createContext<boolean | undefined>(undefined);
+
+export const RowSelectionProvider = RowSelectionContext.Provider;
+
+function useSelection() {
+  const context = useContext(RowSelectionContext);
+  if (context === undefined) {
+    throw new Error('useRowSelection must be used within a RowSelectionContext');
+  }
+  return context;
+}
+
+export const RowSelectionChangeContext =
+  createContext<((selectRowEvent: SelectRowEvent) => void) | undefined>(undefined);
+
+export const RowSelectionChangeProvider = RowSelectionChangeContext.Provider;
+
+function useRowSelectionChange() {
+  const context = useContext(RowSelectionChangeContext);
+  if (context === undefined) {
+    throw new Error('useRowSelectionChange must be used within a RowSelectionChangeContext');
+  }
+  return context;
+}
+
+export function useRowSelection(): readonly [boolean, (selectRowEvent: SelectRowEvent) => void] {
+  return [useSelection(), useRowSelectionChange()];
```

Comment:
Let's inline all this

we can have a single `if` check
```js
if (context1 === undefined || context2 === undefined) throw ....
```

I'd also change the error message to be `'useRowSelection must be used within DataGrid cells'` or something like that.
We don't expose the providers themselves, so the error message wouldn't be very useful to devs otherwise.